src/main/java/org/zalando/planb/revocation/api/impl/RevocationResourceImpl.java

```diff
@@ -94,18 +117,40 @@ public RevocationList get(@RequestParam(required = true) final int from) {
         return responseBody;
     }
 
+    /**
+     * Posts the specified revocation to be stored.
+     * <p>
+     * <p>Revokes tokens associated with the specified revocation type.</p>
+     * <p>
+     * <p>If the field {@code issued_before} is a timestamp set in the future, returns {@link HttpStatus#BAD_REQUEST}.
+     * </p>
+     *
+     * @param revocation the revocation associated with the tokens to revoke
+     */
     @Override
     @RequestMapping(method = RequestMethod.POST)
     @ResponseStatus(HttpStatus.CREATED)
-    public HttpEntity<String> post(@RequestBody final RevocationData revocation) {
-        // don't use revocation data here
-        if (storage.storeRevocation(revocation)) {
-
-            // TODO Refactor
-            return new ResponseEntity<>(HttpStatus.CREATED);
-        } else {
-            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+    public void post(@RequestBody final RevocationRequest revocation) {
+
+        Integer timestamp = null;
+        switch (revocation.getType()) {
+            case TOKEN:
+                timestamp = ((RevokedTokenData) revocation.getData()).getIssuedBefore();
+                break;
+            case CLAIM:
+                timestamp = ((RevokedClaimsData) revocation.getData()).getIssuedBefore();
+                break;
+            case GLOBAL:
+                timestamp = ((RevokedGlobal) revocation.getData()).getIssuedBefore();
+                break;
         }
+
+        // Checks for future timestamps
+        if (timestamp > UnixTimestamp.now() + revocationProperties.getTimestampThresold()) {
```

Comment:
It does, when the object is created, see https://github.com/zalando/planb-revocation/blob/master/src/main/java/org/zalando/planb/revocation/domain/RevokedClaimsData.java#L27